Use CM Console msi's file table for version check

theaquamarine opened the pull request:
The CM Console installer's versions are not changed for hotfixes, even if the installed Console's version changes, so use the extracted contents of the installer for version checking.

Performing an administrative install of the msi is a fairly horrible way to achieve this but 7za can't expand msis usefully (7z.exe can).

DanGough commented:
Would it be better to read the file version directly from the file table in the MSI?

theaquamarine commented:
Yes, that's a massively better idea, nice one!

DanGough commented:
There are plenty of examples online showing how to read MSI tables via the COM interface. I prefer to wrap the C# dlls that come with WiX in my own Powershell projects, but it does add complications of relying on an external dll. Shout if you need an assist!

theaquamarine commented:
Added a function to get the version of a file from an MSI's file table, and changed to using that for the CM console's version. Decided to use the `WindowsInstaller.Installer` COM object to do it to not add any dependencies.
